packages/dashboard/src/components/dashboard/main-menu.tsx

```diff
@@ -52,55 +90,235 @@ export const MainMenu = React.memo((props: MainMenuProps) => {
+      summary: { operational: modeCounter.operational, outOfOrder: modeCounter.outOfOrder },
+      outOfOrder: spoiltEquipment,
+    };
+  };
+
+  const getDispenserSummary = (): ItemSummary => {
+    let modeCounter = { operational: 0, outOfOrder: 0 };
+    const spoiltEquipment: string[] = [];
+    const dispensers = itemState.dispensers;
+    const dispenserKeys = Object.keys(dispensers);
+
+    dispenserKeys.forEach((dispenser) => {
+      switch (dispensers[dispenser].mode) {
+        case RomiCore.DispenserState.IDLE:
+        case RomiCore.DispenserState.BUSY:
+        case RomiCore.DispenserState.OFFLINE:
+          modeCounter['operational'] += 1;
+          break;
+        default:
+          spoiltEquipment.push(dispenser + '- Unknown');
+          modeCounter['outOfOrder'] += 1;
+          break;
+      }
+    });
+
+    return {
+      item: 'Dispensers',
+      summary: { operational: modeCounter.operational, outOfOrder: modeCounter.outOfOrder },
+      outOfOrder: spoiltEquipment,
+    };
+  };
+
+  const getRobotSummary = (): ItemSummary => {
+    let modeCounter = { operational: 0, outOfOrder: 0, charging: 0, idle: 0 };
+    const spoiltEquipment: string[] = [];
+    const fleets = itemState.robots;
+    const fleetKeys = Object.keys(fleets);
+
+    fleetKeys.forEach((fleet) => {
+      fleets[fleet].robots.forEach((robot) => {
+        switch (robot.mode.mode) {
+          case RomiCore.RobotMode.MODE_ADAPTER_ERROR:
+          case RomiCore.RobotMode.MODE_EMERGENCY:
+            spoiltEquipment.push(robot.name + ` - ${robotModeToString(robot.mode)}`);
+            modeCounter['outOfOrder'] += 1;
+            break;
+          case RomiCore.RobotMode.MODE_CHARGING:
+            modeCounter['operational'] += 1;
+            modeCounter['charging'] += 1;
+            break;
+          case RomiCore.RobotMode.MODE_DOCKING:
+          case RomiCore.RobotMode.MODE_GOING_HOME:
+          case RomiCore.RobotMode.MODE_MOVING:
+          case RomiCore.RobotMode.MODE_PAUSED:
+          case RomiCore.RobotMode.MODE_WAITING:
+            modeCounter['operational'] += 1;
+            break;
+          case RomiCore.RobotMode.MODE_IDLE:
+            modeCounter['operational'] += 1;
+            modeCounter['idle'] += 1;
+            break;
+        }
+      });
+    });
+    return {
+      item: 'Robots',
+      summary: {
+        operational: modeCounter.operational,
+        outOfOrder: modeCounter.outOfOrder,
+        charging: modeCounter.charging,
+        idle: modeCounter.idle,
+      },
+      outOfOrder: spoiltEquipment,
+    };
+  };
+
+  const getAllEquipmentSummary = () => {
+    const getDoor = getDoorSummary();
+    const getLift = getLiftSummary();
+    const getDispensers = getDispenserSummary();
+    const getRobots = getRobotSummary();
+    return {
+      door: getDoor,
+      lift: getLift,
+      dispenser: getDispensers,
+      robot: getRobots,
+    };
+  };
+
+  const toggleBannerColor = (): boolean => {
```

Comment:
Is seems misleading, it's called `toggleBannerColor` but doesn't actually toggle anything and returns a boolean.